src/main/plots/solution_graph_statistics_plots.py

```diff
@@ -0,0 +1,107 @@
+                              format: consts.EXTENSION = consts.EXTENSION.HTML, auto_open: bool = False,
+                              x_category_order: PLOTTY_CATEGORY_ORDER = PLOTTY_CATEGORY_ORDER.CATEGORY_ASCENDING) -> None:
+    statistics_df = pd.DataFrame(statistics_dict.items(), columns=[column.value, STATISTICS_FREQ])
+    # x_category_order='total ascending' means: in order of increasing values in Y
+    # x_category_order='category ascending' means: in order of increasing values in X
+    fig = px.bar(statistics_df, x=column.value, y=STATISTICS_FREQ, title=title, labels=labels,
+                 hover_data=[column.value, STATISTICS_FREQ])
+    fig.update_layout(
+        yaxis=dict(
+            title_text=STATISTICS_SHOWING_KEY.FREQ.value
+        ),
+        xaxis=dict(
+            title_text=get_readable_key(column.value),
+            categoryorder=x_category_order.value
+        ),
+        plot_bgcolor=STATISTICS_COLORS.BAR_CHART_BG.value
+    )
+    fig.update_yaxes(automargin=True)
+    save_plot(fig, path, PLOT_TYPE.BAR, plot_name, format, auto_open)
+
+
+def plot_general_statistics(solution_graph: SolutionGraph, path: str = DEFAULT_PATH_FOR_PLOTS,
+                            format: consts.EXTENSION = consts.EXTENSION.HTML, auto_open: bool = False,
+                            x_category_order: PLOTTY_CATEGORY_ORDER = PLOTTY_CATEGORY_ORDER.TOTAL_ASCENDING) -> None:
+    canon_trees_statistics, anon_trees_statistics = get_general_solution_graph_statistics(solution_graph)
+    labels = get_labels_for_freq_plots(STATISTICS_KEY.NODES_NUMBER)
+    __plot_general_statistics(canon_trees_statistics, title='Vertices number for canon trees', path=path,
+                              plot_name='vertices_number_for_canon_trees', format=format, auto_open=auto_open,
+                              x_category_order=x_category_order, labels=labels)
+    __plot_general_statistics(anon_trees_statistics, title='Vertices number for anon trees', path=path,
+                              plot_name='vertices_number_for_anon_trees', format=format, auto_open=auto_open,
+                              x_category_order=x_category_order, labels=labels)
+
+
+def __add_canon_nodes_numbers(fig: go.Figure, canon_nodes_number: int,
+                              anon_nodes_number_freq: Dict[int, int]) -> go.Figure:
+    fig.add_trace(
+        go.Scatter(
+            mode='markers',
+            x=[canon_nodes_number],
+            y=[anon_nodes_number_freq.get(canon_nodes_number, 1) / 2],
+            marker=dict(
+                color='White',
+                size=20,
+                line=dict(width=1, color='DarkSlateGrey')
+            ),
+            showlegend=False,
+            marker_symbol='x',
+            hoverinfo='skip'
+        )
+    )
+    return fig
+
+
+def plot_statistics_for_vertex(vertex_id: int, labels: Dict[str, str], anon_nodes_numbers: List[int],
```

Comment:
How about 'plot_node_numbers_freq_for_vertex'? Or maybe it's too long, but I'm afraid that otherwise, we can forget about what does this function plot after some time